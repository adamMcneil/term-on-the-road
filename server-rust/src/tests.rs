use std::collections::HashMap;

use crate::{rocket, types::PlayerData, Answer, Game, };
use rocket::{http::Status, local::Client};

#[test]
fn not_found() {
    let client = Client::new(rocket(None)).unwrap();
    let res = client.get("/this_is_a_bad_request").dispatch();
    assert_eq!(res.status(), Status::NotFound);
}

#[test]
fn heartbeat() {
    let client = Client::new(rocket(None)).unwrap();
    let res = client.get("/api/v1/heartbeat").dispatch();
    assert_eq!(res.status(), Status::Ok);
}

#[test]
fn simple_game() {
    let client = Client::new(rocket(None)).unwrap();
    // Create game as p1
    let p = PlayerData::new("p1");
    let res = client
        .put("/api/v1/game/my_game")
        .body(serde_json::to_string(&p).unwrap())
        .dispatch();
    assert_eq!(res.status(), Status::Ok);
    // Join
    let p = PlayerData::new("p2");
    let res = client
        .post("/api/v1/game/my_game")
        .body(serde_json::to_string(&p).unwrap())
        .dispatch();
    assert_eq!(res.status(), Status::Ok);
    // P2 answers
    let a = Answer::new("p2", "test");
    let res = client
        .post("/api/v1/game/my_game/answer")
        .body(serde_json::to_string(&a).unwrap())
        .dispatch();
    assert_eq!(res.status(), Status::Ok);
    // P1 answers
    let a = Answer::new("p1", "sssss");
    let res = client
        .post("/api/v1/game/my_game/answer")
        .body(serde_json::to_string(&a).unwrap())
        .dispatch();
    assert_eq!(res.status(), Status::Ok);
    // Get the state of the game
    let mut res = client.get("/api/v1/game/my_game").dispatch();
    let game = serde_json::from_str::<Game>(&res.body_string().unwrap()).unwrap();
    assert_eq!(game.rounds.len(), 2);
    assert_eq!(game.previous_round().unwrap().player_one_answer, Some("sssss".to_string()));
    assert_eq!(game.previous_round().unwrap().player_two_answer, Some("test".to_string()));
    insta::assert_debug_snapshot!(game);

}