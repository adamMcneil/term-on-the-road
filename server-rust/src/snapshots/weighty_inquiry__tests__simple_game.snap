---
source: src/tests.rs
expression: game
---
Game {
    player_one: Some(
        "p1",
    ),
    player_two: Some(
        "p2",
    ),
    rounds: [
        Round {
            player_one_question: "question 1",
            player_two_question: "question 2",
            player_one_answer: Some(
                "something",
            ),
            player_two_answer: Some(
                "test",
            ),
        },
        Round {
            player_one_question: "question_one",
            player_two_question: "question_two",
            player_one_answer: None,
            player_two_answer: None,
        },
    ],
    board: Board {
        board: {
            'b': 0,
            'c': 0,
            'd': 0,
            'f': 0,
            'g': 1,
            'h': 1,
            'j': 0,
            'k': 0,
            'l': 0,
            'm': 1,
            'n': 1,
            'p': 0,
            'q': 0,
            'r': 0,
            's': 0,
            't': -1,
            'v': 0,
            'w': 0,
            'x': 0,
            'y': 0,
            'z': 0,
        },
        player_one_captured: {},
        player_two_captured: {},
    },
}
